### PR TITLE
Conversion geojson -> geo

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

File: src/conversion.rs [hunk 1]
```diff
@@ -71,6 +71,69 @@ fn create_multi_polygon_type<T>(multi_polygon: &geo::MultiPolygon<T>) -> Vec<Pol
...
+
+fn create_geo_multi_line_string<T>(multi_line_type: &Vec<LineStringType>) -> geo::MultiLineString<T>
+    where T: Float
+{
+    geo::MultiLineString(multi_line_type.iter()
+        .map(|point_type| create_geo_line_string(&point_type))
+        .collect())
+}
+
+fn create_geo_polygon<T>(polygon_type: &PolygonType) -> geo::Polygon<T>
```
Comment (datanel) [inline src/conversion.rs]: good to know

File: src/conversion.rs [hunk 2]
```diff
@@ -71,6 +71,69 @@ fn create_multi_polygon_type<T>(multi_polygon: &geo::MultiPolygon<T>) -> Vec<Pol
...
+        .collect())
+}
+
+impl<T> Into<geo::Point<T>> for geometry::Value
+    where T: Float
+{
+    fn into(self) -> geo::Point<T> {
+        match self {
+            geometry::Value::Point(point_type) => create_geo_point(&point_type),
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 3]
```diff
@@ -81,6 +144,21 @@ impl<'a, T> From<&'a geo::Point<T>> for geometry::Value
...
+    where T: Float
+{
+    fn into(self) -> geo::MultiPoint<T> {
+        match self {
+            geometry::Value::MultiPoint(multi_point_type) => {
+                geo::MultiPoint(multi_point_type.iter()
+                    .map(|point_type| create_geo_point(&point_type))
+                    .collect())
+            }
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 4]
```diff
@@ -94,6 +172,19 @@ impl<'a, T> From<&'a geo::MultiPoint<T>> for geometry::Value
...
 
+impl<T> Into<geo::LineString<T>> for geometry::Value
+    where T: Float
+{
+    fn into(self) -> geo::LineString<T> {
+        match self {
+            geometry::Value::LineString(ref multi_point_type) => {
+                create_geo_line_string(multi_point_type)
+            }
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 5]
```diff
@@ -104,6 +195,19 @@ impl<'a, T> From<&'a geo::LineString<T>> for geometry::Value
...
 
+impl<T> Into<geo::MultiLineString<T>> for geometry::Value
+    where T: Float
+{
+    fn into(self) -> geo::MultiLineString<T> {
+        match self {
+            geometry::Value::MultiLineString(ref multi_line_string_type) => {
+                create_geo_multi_line_string(multi_line_string_type)
+            }
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 6]
```diff
@@ -114,6 +218,17 @@ impl<'a, T> From<&'a geo::MultiLineString<T>> for geometry::Value
     }
 }
 
+impl<T> Into<geo::Polygon<T>> for geometry::Value
+    where T: Float
+{
+    fn into(self) -> geo::Polygon<T> {
+        match self {
+            geometry::Value::Polygon(ref polygon_type) => create_geo_polygon(polygon_type),
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 7]
```diff
@@ -124,6 +239,19 @@ impl<'a, T> From<&'a geo::Polygon<T>> for geometry::Value
...
 
+impl<T> Into<geo::MultiPolygon<T>> for geometry::Value
+    where T: Float
+{
+    fn into(self) -> geo::MultiPolygon<T> {
+        match self {
+            geometry::Value::MultiPolygon(ref multi_polygon_type) => {
+                create_geo_multi_polygon(multi_polygon_type)
+            }
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 8]
```diff
@@ -134,22 +262,49 @@ impl<'a, T> From<&'a geo::MultiPolygon<T>> for geometry::Value
...
+                    .map(|geometry| geometry.value.clone().into())
+                    .collect();
+
+                geo::GeometryCollection(geojson_geometries)
             }
-            geo::Geometry::Polygon(ref polygon) => geometry::Value::from(polygon),
-            geo::Geometry::MultiPolygon(ref multi_polygon) => geometry::Value::from(multi_polygon),
-            geo::Geometry::GeometryCollection(ref geometry_collection) => {
-                geometry::Value::from(geometry_collection)
+            _ => panic!("Not valid value {:?}", self),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

File: src/conversion.rs [hunk 9]
```diff
@@ -134,22 +262,49 @@ impl<'a, T> From<&'a geo::MultiPolygon<T>> for geometry::Value
...
+            geometry::Value::MultiLineString(ref multi_line_string_type) => {
+                geo::Geometry::MultiLineString(create_geo_multi_line_string(multi_line_string_type))
             }
+            geometry::Value::Polygon(ref polygon_type) => {
+                geo::Geometry::Polygon(create_geo_polygon(polygon_type))
+            }
+            geometry::Value::MultiPolygon(ref multi_polygon_type) => {
+                geo::Geometry::MultiPolygon(create_geo_multi_polygon(multi_polygon_type))
+            }
+            _ => panic!("GeometryCollection not allowed"),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

But this case is quite complicated as only one variant can't be converted

File: src/conversion.rs [hunk 10]
```diff
@@ -167,6 +322,24 @@ impl<'a, T> From<&'a geo::GeometryCollection<T>> for geometry::Value
...
+        match *geometry {
+            geo::Geometry::Point(ref point) => geometry::Value::from(point),
+            geo::Geometry::MultiPoint(ref multi_point) => geometry::Value::from(multi_point),
+            geo::Geometry::LineString(ref line_string) => geometry::Value::from(line_string),
+            geo::Geometry::MultiLineString(ref multi_line_string) => {
+                geometry::Value::from(multi_line_string)
+            }
+            geo::Geometry::Polygon(ref polygon) => geometry::Value::from(polygon),
+            geo::Geometry::MultiPolygon(ref multi_polygon) => geometry::Value::from(multi_polygon),
+            _ => panic!("GeometryCollection not allowed"),
```
Comment (TeXitoi) [inline src/conversion.rs]: Nop: https://doc.rust-lang.org/std/convert/trait.Into.html => Note: this trait must not fail

But this case is quite complicated as only one variant can't be converted

File: src/conversion.rs (hunk 3 above)
Comment (TeXitoi) [inline src/conversion.rs]: I don't have any solution, but implementing Into seems buggy.

Comment (TeXitoi) [inline src/conversion.rs]: I think a free function while waiting to `try_into` is better.

File: src/conversion.rs (hunk 1 above)
Comment (datanel) [inline src/conversion.rs]: :ok_hand:

**Pull request closed without merging**